src/vdom/component.js

```diff
@@ -17,39 +17,67 @@ import { removeNode } from '../dom/index';
  *	@param {boolean} [opts.render=true]			If `false`, no render will be triggered.
  */
 export function setComponentProps(component, props, opts, context, mountAll) {
-	if (component._disable) return;
-	component._disable = true;
 
-	if ((component.__ref = props.ref)) delete props.ref;
-	if ((component.__key = props.key)) delete props.key;
+	const requestIdleCallback =
+      typeof window !== 'undefined' &&
+      typeof window.requestIdleCallback !== 'undefined' ?
+         window.requestIdleCallback :
+         null;
 
-	if (!component.base || mountAll) {
-		if (component.componentWillMount) component.componentWillMount();
-	}
-	else if (component.componentWillReceiveProps) {
-		component.componentWillReceiveProps(props, context);
-	}
+	const cancelIdleCallback =
+      requestIdleCallback !== 'null' ?
```

Comment:
@makingoff is right, `requestIdleCallback` will be `null` not `'null'` when `window.requestIdleCallback` is not defined, so `cancelIdleCallback` will be `undefined` instead of `null` in that case.

<img width="483" alt="screen shot 2016-12-02 at 3 47 44 pm" src="https://cloud.githubusercontent.com/assets/6886061/20851551/cd7a4a30-b8a6-11e6-9ab2-d4b0208f929a.png">


It doesn't actually cause an error since `cancelIdleCallback` is never called when `requestIdleCallback` is not defined, but still worth fixing 😄